tests/models/llama/test_modeling_llama.py

```diff
@@ -654,6 +654,325 @@ def decode_one_tokens(model, cur_token, input_pos, cache_position):
+gaponcodingbenchmarks. Llama 2 70BresultsareonparorbetterthanPaLM(540B)(Chowdheryetal.,
+2022)onalmostallbenchmarks. Thereisstillalargegapinperformancebetween Llama 2 70BandGPT-4
+and PaLM-2-L.
+We also analysed the potential data contamination and share the details in Section A.6.
+Benchmark (shots) GPT-3.5 GPT-4 PaLM PaLM-2-L Llama 2
+MMLU (5-shot) 70.0 86.4 69.3 78.3 68.9
+TriviaQA (1-shot) – – 81.4 86.1 85.0
+Natural Questions (1-shot) – – 29.3 37.5 33.0
+GSM8K (8-shot) 57.1 92.0 56.5 80.7 56.8
+HumanEval (0-shot) 48.1 67.0 26.2 – 29.9
+BIG-Bench Hard (3-shot) – – 52.3 65.7 51.2
+Table 4: Comparison to closed-source models on academic benchmarks. Results for GPT-3.5 and GPT-4
+are from OpenAI (2023). Results for the PaLM model are from Chowdhery et al. (2022). Results for the
+PaLM-2-L are from Anil et al. (2023).
+
+== END ARTICLE ==
+
+'''
+        question = "What is the paper about?"
+        model_inputs = tokenizer(VERY_LONG_INPUT + question, return_tensors="pt").to(torch_device)
+
+        # No RoPE scaling -> garbage output
+        model = LlamaForCausalLM.from_pretrained(
+            "meta-llama/Llama-2-7b-hf", device_map="auto", load_in_4bit=True,
+        )
+        self.assertTrue(model_inputs["input_ids"].shape[1] > model.config.max_position_embeddings)
+        generate_kwargs = {"max_new_tokens": 40, "do_sample": False}
+        gen_out = model.generate(**model_inputs, **generate_kwargs)
+        decoded_text = tokenizer.decode(gen_out[0], skip_special_tokens=True)
+        self.assertTrue(decoded_text.endswith("Ћ\nЋ\nЋЋЋЋЋЋЋЋЋ\nЋ\nЋ\n"))
+
+        # Dynamic NTK RoPE scaling -> good output (doesn't need fine-tuning)
+        model = LlamaForCausalLM.from_pretrained(
+            "meta-llama/Llama-2-7b-hf", device_map="auto", load_in_4bit=True, rope_scaling={"type": "dynamic", "factor": 2.0},
+        )
+        generate_kwargs = {"max_new_tokens": 40, "do_sample": False}
+        gen_out = model.generate(**model_inputs, **generate_kwargs)
+        decoded_text = tokenizer.decode(gen_out[0], skip_special_tokens=True)
+        self.assertTrue(decoded_text.endswith("The paper is about the release of Llama 2, a family of pretrained and fine-tuned large language models.\nWhat is Llama 2?\n"))
+        # Note: the output above matches our initial release of RoPE scaling
+
+        # Linear RoPE scaling -> usualy okay output (should be used with fine-tuning)
+        model = LlamaForCausalLM.from_pretrained(
+            "meta-llama/Llama-2-7b-hf", device_map="auto", load_in_4bit=True, rope_scaling={"type": "linear", "factor": 2.0},
+        )
+        generate_kwargs = {"max_new_tokens": 40, "do_sample": False}
+        gen_out = model.generate(**model_inputs, **generate_kwargs)
+        decoded_text = tokenizer.decode(gen_out[0], skip_special_tokens=True)
+        self.assertTrue(decoded_text.endswith("The paper is about the development of Llama 2, a large language model (LLM) family, and the release of Llama 2-Chat, a fine-"))
```

Comment:
This assertion would fail if #29765's suspicion [that we were applying the scaling factor twice] was true.

All these text terminations match the output on our original RoPE scaling release, meaning there was no regression since then :)